Add distrib test utilities for c++

yang-g opened the pull request.

jtattermusch commented:
Started the artifact-package-distribtest flow for your branch here:
https://grpc-testing.appspot.com/view/Artifacts/job/gRPC_build_artifacts/885/

jtattermusch commented:
LGTM for the changes, let's wait for the distribtest result to see if it's passing.

jtattermusch commented on tools/dockerfile/distribtest/cpp_jessie_x64/Dockerfile:
```diff
@@ -0,0 +1,53 @@
+      pkg-config \
+      unzip && apt-get clean
+
+RUN git clone https://github.com/grpc/grpc.git
```
Ah, this is actually wrong. The dockerimage is just the base image and should not contain any gRPC sources. 

Then, in the run_distrib_test.sh script, you need to grab the sources of a specific commit that is getting tested   and use those.

jtattermusch commented:
Undo LGTM.

jtattermusch commented:
Did you chmod u+x  the script?
```
/var/local/jenkins/grpc/tools/run_tests/dockerize/docker_run.sh: line 54: test/distrib/cpp/run_distrib_test.sh: Permission denied
```

yang-g commented:
I think I changed the file while you changed the language from c++ to cpp.
Then I added +x. At least we have a different error now :)

On Wed, May 25, 2016 at 1:51 PM, Jan Tattermusch <notifications@github.com>
wrote:

> Did you chmod u+x the script?
>
> /var/local/jenkins/grpc/tools/run_tests/dockerize/docker_run.sh: line 54: test/distrib/cpp/run_distrib_test.sh: Permission denied
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc/pull/6704#issuecomment-221651749>
>

yang-g commented:
Updated according to comments. PTAL

Seems to pass a manual run:
https://grpc-testing.appspot.com/view/Artifacts/job/gRPC_distribtest/architecture=x64,language=cpp,platform=linux/860/

yang-g commented:
Ping~

jtattermusch commented on test/distrib/cpp/run_distrib_test.sh:
```diff
@@ -0,0 +1,52 @@
+
+set -ex
+
+git clone https://github.com/grpc/grpc.git
```
if you run this e.g. on the release branch, it still runs the distrib tests on upstream/master. 

We need to clone the repo for outside of the docker container.

jtattermusch replied:
This might require a few more changes, as we are actually doing the cloning trick when building artifacts, but for distrib tests, we only copy contents of test/distrib directory from outside the container (to make sure we don't copy more then we need for the tests).

yang-g commented:
Now I am cloning from the local copy. Verified with a one off test that the local change is picked up. PTAL thanks.

jtattermusch commented on tools/dockerfile/distribtest/cpp_jessie_x64/Dockerfile:
```diff
@@ -0,0 +1,45 @@
+
+FROM debian:jessie
+
+# Install packages for gRPC and protobuf
```
nit: the comment is not up-to-date

jtattermusch commented on test/distrib/cpp/run_distrib_test.sh:
```diff
@@ -0,0 +1,52 @@
+
+set -ex
+
+git clone $EXTERNAL_GIT_ROOT
```
you could do just --recursive so you don't need git submodule update, but this works as well.

jtattermusch commented on test/distrib/cpp/run_distrib_test.sh:
```diff
@@ -0,0 +1,52 @@
+cd grpc && git submodule update --init
+
+cd third_party/protobuf && ./autogen.sh && \
+./configure && make -j12 && make check && make install && ldconfig
```
nit: -j12 is probably too much (multiple distribtests are running in parallel on the same worker).  -j8 or -j4 should be fine.

jtattermusch commented:
LGTM with some minor nits.

yang-g commented:
All addressed.
